presto/presto.sh

```diff
@@ -12,75 +12,106 @@
 #    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 #    See the License for the specific language governing permissions and
 #    limitations under the License.
-set -x -e
+
+set -euxo pipefail
 
 # Variables for running this script
-ROLE=$(/usr/share/google/get_metadata_value attributes/dataproc-role)
-HOSTNAME=$(hostname -s)
-PRESTO_MASTER_FQDN=$(/usr/share/google/get_metadata_value attributes/dataproc-master)
-WORKER_COUNT=$(/usr/share/google/get_metadata_value attributes/dataproc-worker-count)
-CONNECTOR_JAR=$(find /usr/lib/hadoop/lib -name 'gcs-connector-*.jar')
-PRESTO_VERSION="0.177"
-HTTP_PORT="8080"
-
-# Download and unpack Presto server
-wget https://repo1.maven.org/maven2/com/facebook/presto/presto-server/${PRESTO_VERSION}/presto-server-${PRESTO_VERSION}.tar.gz
-tar -zxvf presto-server-${PRESTO_VERSION}.tar.gz
-mkdir /var/presto
-mkdir /var/presto/data
-
-# Copy required Jars
-cp ${CONNECTOR_JAR} presto-server-${PRESTO_VERSION}/plugin/hive-hadoop2
-
-# Configure Presto
-mkdir presto-server-${PRESTO_VERSION}/etc
-mkdir presto-server-${PRESTO_VERSION}/etc/catalog
-
-cat > presto-server-${PRESTO_VERSION}/etc/node.properties <<EOF
+readonly ROLE="$(/usr/share/google/get_metadata_value attributes/dataproc-role)"
+readonly PRESTO_MASTER_FQDN="$(/usr/share/google/get_metadata_value attributes/dataproc-master)"
+readonly WORKER_COUNT=$(/usr/share/google/get_metadata_value attributes/dataproc-worker-count)
+readonly CONNECTOR_JAR="$(find /usr/lib/hadoop/lib -name 'gcs-connector-*.jar')"
+readonly PRESTO_VERSION='0.177'
+readonly HTTP_PORT='8080'
+readonly INIT_SCRIPT='/usr/lib/systemd/system/presto.service'
+PRESTO_JVM_MB=0;
+PRESTO_QUERY_NODE_MB=0;
+PRESTO_RESERVED_SYSTEM_MB=0;
+
+function err() {
+  echo "[$(date +'%Y-%m-%dT%H:%M:%S%z')]: $@" >&2
+  return 1
+}
+
+function get_presto(){
+  # Download and unpack Presto server
+  wget https://repo1.maven.org/maven2/com/facebook/presto/presto-server/${PRESTO_VERSION}/presto-server-${PRESTO_VERSION}.tar.gz
+  tar -zxvf presto-server-${PRESTO_VERSION}.tar.gz
+  mkdir -p /var/presto/data
+}
+
+function calculate_memory(){
```

Comment:
Ah, sorry to make you refactor such an ugly function. There are easier ways to calculate memory on a node that don't involve spark. For example, just `bdconfig get_property_value --configuration_file /etc/hadoop/conf/yarn-site.xml --name yarn.nodemanager.resource.memory-mb` would tell you how much memory is configured on a worker node manager. This number is 80% of the node's total memory (in MB).

I think you can use that number to calculate the 60%/40% thing below. Presto's instructions for configuring memory are here: https://prestodb.io/presto-admin/docs/current/installation/presto-configuration.html.

If you want to do this in a different PR, feel free to push back.

Reply:
I'll be happy to do it, but I suggest another PR as this isn't just a style change